Rx/v2/src/rxcpp/operators/rx-retry.hpp

```diff
@@ -21,165 +21,47 @@
-                                  state->do_subscribe();
-                                }
-                              },
-                              // on_completed
-                              [state]() {
-                                // JEP: never appeears to be called?
-                                state->out.on_completed();
-                              }
-                              );
-    }
-    
-    composite_subscription source_lifetime;
-    output_type out;
-    composite_subscription::weak_subscription lifetime_token;
-  };
-
-  // Finite retry case (explicitly limited with the number of times)
-  template<class T, class Observable, class Count>
-  struct finite : public operator_base<T> {
-    typedef rxu::decay_t<Observable> source_type;
-    typedef rxu::decay_t<Count> count_type;
-
-    struct values {
-      values(source_type s, count_type t)
-        : source(std::move(s)), remaining_(std::move(t)) {
-      }
-      
-      inline bool completed_predicate() const {
-        // Return true if we are completed
-        return remaining_ <= 0;
-      }
-      
-      inline void update() {
-        // Decrement counter
-        --remaining_;
-      }
-
-      source_type source;
-
-    private:
-      count_type remaining_;
-    };
-
-    finite(source_type s, count_type t)
-      : initial_(std::move(s), std::move(t)) {
-    }
-
-    template<class Subscriber>
-    void on_subscribe(const Subscriber& s) const {
-      typedef state_type<values, Subscriber, T> state_t;
-      // take a copy of the values for each subscription
-      auto state = std::make_shared<state_t>(initial_, s);
-      if (initial_.completed_predicate()) {
-        // Should we call it here in retry?
-        state->out.on_completed();
+  struct event_handlers {
+    template <typename State>
+    static inline void on_error(State& state, std::exception_ptr& e) {
+      state->update();
+      // Use specialized predicate for finite/infinte case
+      if (state->completed_predicate()) {
+        state->out.on_error(e);                                  
       } else {
-        // start the first iteration
         state->do_subscribe();
       }
     }
-
-  private:
-    values initial_;
-  };
-
-  // Infinite retry case
-  template<class T, class Observable>
-  struct infinite : public operator_base<T> {
-    typedef rxu::decay_t<Observable> source_type;
-
-    struct values {
-      values(source_type s)
-        : source(std::move(s)) {
-      }
-      
-      static inline bool completed_predicate() {
-        // Infinite retry never stops ignoring errors
-        return false;
-      }
-      
-      static inline void update() {
-        // Infinite retry does not need to update state
-      }
-
-      source_type source;
-    };
-
-    infinite(source_type s) : initial_(std::move(s)) {
+          
+    template <typename State>
+    static inline void on_completed(State& state) {
+      // JEP: never appeears to be called?
```

Comment:
nothing to worry about, but I do not think that we need to keep this comment around anymore. :)